src/deferred.js

```diff
@@ -16,12 +16,12 @@ jQuery.extend({
 	Deferred: function( func ) {
 		var tuples = [
 				// action, add listener, callbacks, .then handlers, final state
+				[ "notify", "progress", jQuery.Callbacks("memory"),
```

Comment:
I don't know how much bigger the code would get if this was an array of props, but it would probably make the logic easier to follow. I can live with it tho.